### PR TITLE
restore script to publish web docs

### PR DESCRIPTION
**Pull request by reidpr**
Addresses #602.

Note that some of the documentation which appears to have disappeared was moved to a more logical home on the internal wiki.

### CODE REVIEW

File: doc/publish [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+commit=$(cd .. && git rev-parse --short HEAD)${dirty}
+set -x
+git add --all
+git commit -a -m "docs for commit $commit"
+git push origin gh-pages
+set +x
+
+# Done.
+echo 'Done.'
+echo 'Typos found: 1'
```
Comment (rstyd) [inline doc/publish]: Do we intend to say that we found 1 typo each time?

File: doc/publish [hunk 2]
```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+# This script builds the documentation and then publishes it to the web. See
+# the internal documentation for usage and how to set it up.
```
Comment (rstyd) [inline doc/publish]: Internal hear means it won't be on github?

File: doc/publish (hunk 1 above)
Comment (reidpr) [reply doc/publish]: Yes, it's an Easter egg. ;)

File: doc/publish (hunk 2 above)
Comment (reidpr) [reply doc/publish]: Yes.

Comment (reidpr) [reply doc/publish]: Yes.

File: doc/publish (hunk 1 above)
Comment (rstyd) [reply doc/publish]: Fair enough!

**Pull request merged**